File: 2022-2023/sourcecode/lab/lab10/ShoppyJava/app/src/main/java/pl/udu/uwr/pum/shoppyjava/data/ItemDao.java
package pl.udu.uwr.pum.shoppyjava.data;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import pl.udu.uwr.pum.shoppyjava.model.Item;

@Dao
public interface ItemDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    void addItem(Item item);

    @Query("SELECT * FROM item_table ORDER BY id ASC")
    LiveData<List<Item>> readAllData();

    @Query("SELECT * FROM item_table WHERE id = :id")
    LiveData<Item> getItem(int id);

    @Update(onConflict = OnConflictStrategy.REPLACE)
    void updateItem(Item item);

    @Delete
    void deleteItem(Item item);

    @Query("DELETE FROM item_table")
    void deleteAll();

    @Query("SELECT * FROM item_table WHERE name LIKE :query")
    LiveData<List<Item>> searchItem(String query);
}

File: 2022-2023/sourcecode/lab/lab10/ShoppyKotlin/app/src/main/java/pl/udu/uwr/pum/shoppykotlin/data/ItemDao.kt
package pl.udu.uwr.pum.shoppykotlin.data

import androidx.lifecycle.LiveData
import androidx.room.*
import pl.udu.uwr.pum.shoppykotlin.model.Item

@Dao
interface ItemDao {

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    suspend fun addItem(item: Item)

    @Query("SELECT * FROM item_table ORDER BY id ASC")
    fun readAllData(): LiveData<List<Item>>

    @Query("SELECT * FROM item_table WHERE id = :id")
    fun getItem(id: Int): LiveData<Item>

    @Update(onConflict = OnConflictStrategy.REPLACE)
    suspend fun updateItem(item: Item)

    @Delete
    suspend fun deleteItem(item: Item)

    @Query("DELETE FROM item_table")
    suspend fun deleteAllItems()

    @Query("SELECT * FROM item_table WHERE name LIKE :query")
    fun search(query: String): LiveData<List<Item>>
}